### ISSUE TITLE
Add a more concise syntax / method for responsive breakpoints

### ISSUE THREAD

**Issue by smlarkin** [retyui/react-native-stylex#8]
## Issue

The current implementation of breakpoints allows for a CSS-in-JS approach to media queries.  This is a nice addition to a stylesheet, but can be wet and verbose.

## Suggestion

Add a utility function that evaluates a set of breakpoints setup in the theme so they can be applied to any parameter that can be responsively sized.  These would be a simple subset of of for `maxWidth` (or possibly `minWidth`, etc)...

```
// THEME
const theme = {
  palette: { textColor: "black" },
  utils: { fade: (color, value) => {} },
  breakpoints: [320, 480, 640, 768, 1024, 1200, 1920]
};

// STLYING
import { applyBreakpoints as abs } from "react-native-stylex/media-query"

const useStyles = makeUseStyles(({ palette, utils }) => ({
  root: {
    color: utils.fade(palette.text.textColor, 0.5),
    height: abs([69, 100, 300]),  // each index references the breakpoints index
    width: abs({0: 100, 6: 500}) // an object can be used to skip indexes and style non-adjacent breakpoints
}));
```
## Notes

Breakpoints could optionally be setup with `width` & `height` to follow the current API closer for more control if you need more than just `width`...

```
breakpoints: [320, {width: 480, height 240}, 640, 768]
```

**Comment by retyui** [comment]
@smlarkin Thanks for that interesting concept. Do you have an idea how we can make syntax mode expressive, extendable, safety and human-readable


What do you think about this variant:

```ts
const applyBreakpoints = createApplyBreakpoints({
  xxs: 320,
  xs: 480,
  s: 640,
  m: 768,
  l: 1024,
  xl: 1200,
  xxl: 1920,
});

const useStyles = makeUseStyles(() => ({
  root: {
    fontSize: applyBreakpoints({ xss: 20, xs: 18, default: 16 }),
  },
}));

```

**Comment by smlarkin** [comment]
@retyui your design is probably a safer / cleaner approach than mine when it comes to syntax/semantics.

So yeah, that would work well, actually :)

**Comment by retyui** [comment]
Fixed in [4.1.0](https://github.com/retyui/react-native-stylex/blob/master/CHANGELOG.md#410)

diff: [4.0.0...4.1.0](https://npmfs.com/compare/react-native-stylex/4.0.0/4.1.0/)
